### DISCUSSION TITLE
Repeater give me error when i use it with BelongsToMany relation

### DISCUSSION THREAD

**Discussion by hasankhdev** [filamentphp/filament#7025] [Help]
Hello Guys,

I have this repeater in CompetitionResource

`Repeater::make('products')
                        ->relationship()
                        ->schema([
                            Select::make('product_id')
                                ->label('Product')
                                ->searchable()
                                ->options(Product::all()->pluck('name', 'id')),
                            TextInput::make('quantity')->numeric()->required()
                        ])
                        ->minItems(1)
                        ->maxItems(3)
                        ->createItemButtonLabel('Add product')
                        ->defaultItems(3)
                        ->collapsible()
                        ->collapsed()
                        ->columns(2)
`

The relation between the repeater and the products is belongsToMany:

`
    public function products(): BelongsToMany
    {
        return $this->belongsToMany(Product::class, 'competition_product');
    }
`

When i try to create a new competition, I get an error, because filament trying to add a new product to products table!!!!

But the weird thing when i add the products manually to the competition_product table, its show correctly in the edit competition page.

Again When i trying to edit the quantity or the product, it's try to add new product on the products table.

Kindly, anyone have solution for that.

Thank you.

**Comment by pboivin** [comment]
You need `HasMany` instead of `BelongsToMany` : 
https://github.com/filamentphp/filament/discussions/6731
